tests/src/Functional/SmsFrameworkIncomingBrowserTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Drupal\Tests\sms\Functional;
+
+use Drupal\Core\Url;
+
+/**
+ * Tests manage event access page.
+ *
+ * @group SMS Framework
+ */
+class SmsFrameworkIncomingBrowserTest extends SmsFrameworkBrowserTestBase {
+
+  /**
+   * The HTTP client.
+   *
+   * @var \GuzzleHttp\Client
+   */
+  protected $httpClient;
+
+  /**
+   * An incoming gateway instance.
+   *
+   * @var \Drupal\sms\Entity\SmsGatewayInterface
+   */
+  protected $incomingGateway;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $this->httpClient = $this->container->get('http_client');
+
+    $this->incomingGateway = $this->createMemoryGateway(['plugin' => 'incoming']);
+    $this->incomingGateway
+      ->setSkipQueue(TRUE)
+      ->save();
+    $this->container->get('router.builder')->rebuild();
+  }
+
+  /**
+   * Test incoming route endpoint provided by 'incoming' gateway.
+   */
+  public function testIncomingRouteEndpoint() {
+    $messages[0] = [
+      'message' => $this->randomString(),
+      'recipients' => $this->randomPhoneNumbers(),
+    ];
+    $messages[1] = [
+      'message' => $this->randomString(),
+      'recipients' => $this->randomPhoneNumbers(),
+    ];
+
+    $url = Url::fromRoute('sms.incoming.receive.' . $this->incomingGateway->id())
+      ->setRouteParameter('sms_gateway', $this->incomingGateway->id())
+      ->setAbsolute()
+      ->toString();
+
+    $options = [
+      'json' => [
+        'messages' => $messages,
+      ],
+    ];
+
+    $this->assertTrue(TRUE, sprintf('POST request to %s', $url));
```

Comment:
Its not a test.

Its how the testing framework recommends passing an **informational** line to the test log. 

Like when you use the Simpletest .module to do a `->drupalGet()`, it will say something like "GET /admin/config/smsframework/gateways returned 403 (6.04 KB)." (_\Drupal\simpletest\AssertContentTrait::parse_)

(Technical note: old web tests use `::pass` which passes to `::assertTrue`. A deprecation notice on the browser test base say to use `::assertTrue` directly.)

Yes its weird as hell.